pkg/server/server.go

```diff
@@ -90,6 +93,62 @@ func (s *Server) SetRules(w http.ResponseWriter, r *http.Request, tenant string)
 	_, _ = w.Write([]byte("successfully updated rules file"))
 }
 
+func (s *Server) ListAllRules(w http.ResponseWriter, r *http.Request) {
+	logger := log.With(s.logger, "handler", "listAllRules")
+
+	//nolint:exhaustivestruct
+	allGroups := &rulefmt.RuleGroups{}
+
+	_ = s.bucket.Iter(r.Context(), rulesBasePath, func(dir string) error {
+		tenant := strings.TrimPrefix(dir, rulesBasePath)
+		tenant = strings.TrimSuffix(tenant, "/")
+
+		file, err := s.bucket.Get(r.Context(), getRulesFilePath(tenant))
+		if err != nil {
+			level.Warn(logger).Log("msg", "failed retrieving rules file", "tenant", tenant, "err", err)
+
+			return nil
+		}
+		defer file.Close()
+
+		data, err := ioutil.ReadAll(file)
+		if err != nil {
+			level.Warn(logger).Log("msg", "error reading rules file", "tenant", tenant, "err", err)
+
+			return nil
+		}
+
+		groups, errs := rulefmt.Parse(data)
```

Comment:
Same here, if we can't parse a file that we supposedly wrote, then something is really wrong. The data is likely corrupted and we should respond accordingly

Reply:
In other words, the client asked for all rules, but we are failing to provide all of the rules. We need to indicate this failure to the client